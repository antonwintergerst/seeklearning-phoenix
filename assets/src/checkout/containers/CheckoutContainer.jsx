import React from 'react';

// redux
import { connect } from 'react-redux';
import { goTo } from '../../actions/router';
import { moveToAddAnotherJob } from '../actions';

import OrderSummary from '../components/OrderSummary';
import MockUserCard from '../components/MockUserCard';
import { perks } from '../../home/containers/HomeContainer.data';

const CheckoutContainer = (props) => {
  const { quote, user } = props;
  if (!user) {
    props.goTo('/');
    return null;
  }

  const quoteGroupedByPrice = {
    ...quote,
    products: quote.jobs.reduce((acc, job) => {
      const product = job.adType;
      const exists = acc.find(p => p.id === product.id && p.price === product.price);
      if (exists) {
        return acc.map((p) => {
          if (p.id === product.id && p.price === product.price) {
            return { ...p, quantity: p.quantity + 1 };
          }
          return p;
        });
      }
      return [...acc, { ...product, quantity: 1 }];
    }, []),
  };

  const onContinueClicked = () => {

  };
  const onAddAnotherClicked = () => {
    props.moveToAddAnotherJob();
  };
  return (
    <div className="ui container">
      <h2 className="ui header">Review quote</h2>
      <OrderSummary products={quoteGroupedByPrice.products} />
      {user &&
        <MockUserCard user={user} perks={perks} />
      }
      <button className="ui huge button pink continue" onClick={onContinueClicked}>Continue to payment options</button>
      <button className="ui button" onClick={onAddAnotherClicked}>Add another job</button>
    </div>
  );
};
// Redux state mapping
const mapStateToProps = state => ({
  user: state.home.user,
  quote: state.checkout.quote,
});

export default connect(mapStateToProps, {
  goTo, moveToAddAnotherJob,
})(CheckoutContainer);
