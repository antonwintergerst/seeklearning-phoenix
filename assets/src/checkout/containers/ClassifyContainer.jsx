import React from 'react';
import AdSelectionForm from '../components/AdSelectionForm';
import { adTypes, adFeatures } from './ClassifyContainer.data';

const ClassifyContainer = () => (
  <div>
    <h3 className="ui header">Choose your style of ad</h3>
    <div className="ui stacked segments">
      <AdSelectionForm adTypes={adTypes} adFeatures={adFeatures} />
    </div>
  </div>
);

export default ClassifyContainer;
