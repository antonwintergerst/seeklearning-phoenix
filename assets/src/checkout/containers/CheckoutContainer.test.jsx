import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import CheckoutContainer from './CheckoutContainer';
import { quote } from '../containers/ClassifyContainer.data';

// configure enzyme
configure({ adapter: new Adapter() });

describe('CheckoutContainer', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);

  const initialState = {
    checkout: {
      quote,
    },
    home: {
      user: {
        id: 1,
        name: 'Matt Giampietro',
        occupation: 'Producer, Unilever',
        perks: [1],
        avatar: '/dist/images/matthew.png',
      },
    },
  };

  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<CheckoutContainer store={store} />);
  });

  it('renders', () => {
    expect(container.length).toEqual(1);
  });

  it('maps state to props', () => {
    expect(container.prop('output')).toEqual(initialState.output);
  });

  it('renders as expected', () => {
    const tree = ReactTestRenderer.create(container).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
