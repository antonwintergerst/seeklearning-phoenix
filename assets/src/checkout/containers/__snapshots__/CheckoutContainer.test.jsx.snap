// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutContainer renders as expected 1`] = `
<div
  className="ui container"
>
  <h2
    className="ui header"
  >
    Review quote
  </h2>
  <div
    className="ui basic segment ad-selection-form"
  >
    <table
      className="ui celled striped table"
    >
      <thead
        className="center aligned"
      >
        <tr>
          <th>
            Product
          </th>
          <th>
            Qty
          </th>
          <th>
            Price
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              className="ui ribbon label"
            >
              Classic
            </div>
          </td>
          <td
            className="center aligned"
          >
            1
          </td>
          <td
            className="center aligned"
          >
            <span
              className="price"
            >
              $
              269.99
            </span>
            <span
              className="price-suffix"
            >
              +GST
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="ui ribbon label"
            >
              StandOut
            </div>
          </td>
          <td
            className="center aligned"
          >
            1
          </td>
          <td
            className="center aligned"
          >
            <span
              className="price"
            >
              $
              322.99
            </span>
            <span
              className="price-suffix"
            >
              +GST
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="ui ribbon label"
            >
              Guaranteed Hire
            </div>
          </td>
          <td
            className="center aligned"
          >
            1
          </td>
          <td
            className="center aligned"
          >
            <span
              className="price"
            >
              $
              394.99
            </span>
            <span
              className="price-suffix"
            >
              +GST
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="ui card"
  >
    <div
      className="image"
    >
      <img
        alt=""
        src="/dist/images/matthew.png"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        Matt Giampietro
      </div>
      <div
        className="meta"
      >
        <span>
          Producer, Unilever
        </span>
      </div>
      <div
        className="description"
      >
        <p>
          Gets a for **3 for 2 deal on Classic Ads**
        </p>
      </div>
    </div>
  </div>
  <button
    className="ui huge button pink continue"
    onClick={[Function]}
  >
    Continue to payment options
  </button>
  <button
    className="ui button"
    onClick={[Function]}
  >
    Add another job
  </button>
</div>
`;
